scripts/file_search.py

```diff
@@ -17,48 +17,62 @@ def html_tail():
 
 
 def collect_files(root_folder, ext):
-    '''
+    """
     Crawls in all directories from a list of folders
     and returns all .ext files in any subfolders
-    '''
+    """
     abs_input_file_list = []
     for root, dirs, files in os.walk(root_folder):
         for file in files:
             if file.endswith(ext):
                 abs_input_file_list.append(os.path.join(root, file))
 
-    print('Found '+str(len(abs_input_file_list))+" "+ext+" files")
+    print("Found " + str(len(abs_input_file_list)) + " " + ext + " files")
     return abs_input_file_list
 
 
 def tokenize(text, token, highlight=""):
-    text = text.replace('--\n--\n', '</br>')
-    text = text.replace('\n', '</br>')
+    text = text.replace("--\n--\n", "</br>")
+    text = text.replace("\n", "</br>")
     if highlight:
-        text = text.replace(highlight, '<emph>{}</emph>'.format(highlight))
+        text = text.replace(highlight, "<emph>{}</emph>".format(highlight))
 
     return "<{0:}>{1:}</{0:}>".format(token, text)
 
 
 def parse_and_search(allfiles, todo_token, n_lines_before=5, n_lines_after=5):
     print("Looking for {} in {} files...".format(todo_token, len(allfiles)))
     cnt = 0
-    log_output = 'output.html'
-    with open(log_output, 'w') as f:
+    log_output = "output.html"
+    with open(log_output, "w") as f:
         f.write(html_head())
-        f.write(tokenize("Instances of {}".format(todo_token), 'h1'))
-        f.write(tokenize('Number of lines before : {}'.format(n_lines_before), 'p'))
-        f.write(tokenize('Number of lines after  : {}'.format(n_lines_after), 'p'))
+        f.write(tokenize("Instances of {}".format(todo_token), "h1"))
+        f.write(tokenize("Number of lines before : {}".format(n_lines_before), "p"))
+        f.write(tokenize("Number of lines after  : {}".format(n_lines_after), "p"))
         for file_name in allfiles:
-            p = subprocess.Popen(['egrep', '-i', '-n', '-A{}'.format(n_lines_after), '-B{}'.format(n_lines_before), todo_token, file_name], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+            p = subprocess.Popen(
+                [
+                    "egrep",
+                    "-i",
+                    "-n",
+                    "-A{}".format(n_lines_after),
+                    "-B{}".format(n_lines_before),
+                    todo_token,
+                    file_name,
+                ],
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+            )
```

Comment:
Does this stay like this with longer lines? I'm unpleased to use 12 lines for a `grep`. But so be it.

Reply:
Agreed, it *should* be responsive to line length